cognite_toolkit/cdf_tk/load.py

```diff
@@ -426,6 +427,15 @@ def fixup_resource(local: DataSet, remote: DataSet) -> DataSet:
         local.last_updated_time = remote.last_updated_time
         return local
 
+    def load_resource(self, filepath: Path, ToolGlobals: CDFToolConfig, dry_run: bool) -> DataSetList:
+        resource = load_yaml_inject_variables(filepath, {})
+        data_sets = list(resource) if isinstance(resource, dict) else resource
+        for data_set in data_sets:
+            if data_set.get("metadata"):
+                for key, value in data_set["metadata"].items():
+                    data_set["metadata"][key] = json.dumps(value)
```

Comment:
If `value` is a string this will cause an unnecessary extra `"` around it. 
```suggestion
                    data_set["metadata"][key] = json.dumps(value) if isinstance(value, dict) else value
```